src/framework/uicomponents/view/popupview.cpp

```diff
@@ -211,20 +211,7 @@ void PopupView::open()
         }
 
         qWindow->setTitle(m_title);
-
-        if (m_alwaysOnTop) {
-#ifdef Q_OS_MAC
-            auto updateStayOnTopHint = [this]() {
-                bool stay = qApp->applicationState() == Qt::ApplicationActive;
-                m_window->qWindow()->setFlag(Qt::WindowStaysOnTopHint, stay);
-            };
-            updateStayOnTopHint();
-            connect(qApp, &QApplication::applicationStateChanged, this, updateStayOnTopHint);
-#endif
-        } else {
-            qWindow->setModality(m_modal ? Qt::ApplicationModal : Qt::NonModal);

```

Comment:
The removal of this line causes problems for modal dialogs: now I can switch focus from a warning dialog to the main window!

https://github.com/musescore/MuseScore/assets/48658420/6994b0b2-c088-497f-9d68-1a5cb98afa72